### PR TITLE
AGENT-501: Make it easier to change node zero IP and SERVICE_BASE_URL

### PR DESCRIPTION
**Pull request by rwsu**
The node zero IP and ServiceBaseURL are hard-coded into scripts and
systemd service definition files. This makes it difficult to change
the node zero IP once the agent ISO has booted up. To facilitate
changing these settings in the interactive network configuration
use case, the hard-codings have been replaced with sourcing files
containing environment variables.
    
In the interactive network configuration scenario, the environment 
files will be updated by the network configuration utility. The 
agent services are blocked by the interactive console service. When 
the interactive console service finishes, the agent services startup 
and read in the environment files picking up any changes.
    
In the non-interactive scenario, the environment files will be
pre-populated with values for node zero IP and SERVICE_BASE_URL in
the Agent ISO. This is how it works today.

### CODE REVIEW

File: data/data/agent/files/usr/local/bin/set-node-zero.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# BOOTSTRAP_HOST_MAC is read by assisted-service. The host with a MAC address
+# matching this value in assisted-service is designated to be the bootstrap during
+# cluster installation. In assisted-service.service, this file is included as a
+# --env-file in the ExecStart command.
+#
+# This file is also a ConditionPathExists in the following systemd service
+# definitions:
+# apply-host-config.service
+# assisted-service-pod.service
+# assisted-service.service
```
[the remark this answers is not in the document]
Comment (bfournie) [reply data/data/agent/files/usr/local/bin/set-node-zero.sh:44]: Other than that, looks good. I have not tested it yet but I will.

[the remark this answers is not in the document]
Comment (rwsu) [reply data/data/agent/files/usr/local/bin/set-node-zero.sh:44]: Good catch.

**Pull request merged**